src/re-parse.y

```diff
@@ -212,6 +215,9 @@ ccl		:  ccl TOK_CHAR '-' TOK_CHAR
 				}
 			else
 				$1->Add($2);
+
+			if (zeek::detail::re_single_line && $2 == '.' )
```

Comment:
I don't believe this is correct.  `.` inside a character class is a literal, not a meta-character.

Reply:
Also, I'd expect line 115: 

> 			$$ = new zeek::detail::NFA_Machine(new zeek::detail::NFA_State(zeek::detail::rem->AnyCCL()));

to pass a flag to `AnyCCL` indicating whether it wants the include-newline version of `.` or the don't-include-newline version, with a matching change for `AnyCCL` to create each of those on demand.

Reply:
How should the `EquivClass` in the `Specific_RE_Matcher` be updated in this case then? Do the same thing, using a separate EC for the two modes?

Reply:
I think it should be the same EC for both branches.  ECs span the entirety of the regular expression, so if an RE mixes both types of "any", then the EC needs to be informed of that via `CCL_Use()`.

Reply:
Alright, I think 2d86aa39e630dcbe8ed1b10b4111653ab9cf0cbd covers all of that correctly now.